packages/near-fast-auth-signer/src/components/SocialLogin/SocialButton.tsx

```diff
@@ -69,73 +69,69 @@ function SocialButton({
-        const public_key_lak = decodeIfTruthy(searchParams.get('public_key_lak'));
-        const contract_id = decodeIfTruthy(searchParams.get('contract_id'));
-        const methodNames = decodeIfTruthy(searchParams.get('methodNames'));
-        const accessToken = await user.getIdToken();
-
-        setAccountIdToController({ accountId, networkId });
-
-        let publicKeyFak: string;
-        if (await isPassKeyAvailable()) {
-          const keyPair = await createKey(email);
-          publicKeyFak = keyPair.getPublicKey().toString();
-          await window.fastAuthController.setKey(keyPair);
-        }
-
-        if (!window.fastAuthController.getAccountId()) {
-          await window.fastAuthController.setAccountId(accountId);
-        }
-
-        await window.fastAuthController.claimOidcToken(accessToken);
-        const oidcKeypair = await window.fastAuthController.getKey(`oidc_keypair_${accessToken}`);
-        if (!window.firestoreController) {
-          window.firestoreController = new FirestoreController();
-        }
-        window.firestoreController.updateUser({
-          userUid:   user.uid,
-          oidcToken: accessToken,
-        });
-        const isNewUser = shouldCreateAccount(user.metadata.creationTime, user.metadata.lastSignInTime);
-        let callback;
-        if (isNewUser) {
-          callback = onCreateAccount;
-        } else if (isRecovery) {
-          callback = onSignIn;
-        } else {
-          callback = onCreateAccount;
-        }
-
-        await callback({
-          oidcKeypair,
-          accessToken,
-          accountId,
-          publicKeyFak,
-          public_key_lak,
-          contract_id,
-          methodNames,
-          success_url,
-          setStatusMessage: () => {},
-          navigate,
-          searchParams,
-          gateway:          success_url,
-        });
-      })
-      .catch((error) => {
-        // Handle errors here
-        console.error(error);
+  const onClick = async () => {
+    try {
+      const result = await signInWithPopup(firebaseAuth, provider);
+      const { user } = result;
+
+      // TODO: construct accountId, confirm with UX
+      const accountId = getSocialLoginAccountId();
+      const { email } = user;
+
+      const success_url = decodeIfTruthy(searchParams.get('success_url'));
+      const public_key_lak = decodeIfTruthy(searchParams.get('public_key_lak'));
+      const contract_id = decodeIfTruthy(searchParams.get('contract_id'));
+      const methodNames = decodeIfTruthy(searchParams.get('methodNames'));
+      const accessToken = await user.getIdToken();
+
+      setAccountIdToController({ accountId, networkId });
+
+      const publicKeyFak = await storePassKeyAsFAK(email);
+      if (!window.fastAuthController.getAccountId()) {
+        await window.fastAuthController.setAccountId(accountId);
+      }
+
+      await window.fastAuthController.claimOidcToken(accessToken);
```

Comment:
This still duplicated no? 

https://github.com/near/fast-auth-signer/blob/8d2726f5fe007d7ae011dec245692ac7499aff0c/packages/near-fast-auth-signer/src/components/AuthCallback/AuthCallback.tsx#L248

I think it would be good to extract the common logic from AuthCallback and SocialLogin to a separated component

1 - passkeys
2 - claimoidc
3 - call createAccount or signIn

What do you think? Otherwise we have to maintain the same flow on two places.

Reply:
Also I've extracted some function in the AuthCallback component in my PR, I think we should look for how to reconcile them.

Reply:
I thought previous comment was about refactoring the part that I pull out and replace with `setAccountIdToController`. 

I did notice that @benmalcom also made refactor on this PR, perhaps I will reuse his one. (Either we merge this PR first and once @benmalcom 's [PR](https://github.com/near/fast-auth-signer/pull/264) merged, I refactor or we merge his one first and I will rebase on top of his)

Also the this part is actually different to Authcallback

```
const isNewUser = shouldCreateAccount(user.metadata.creationTime, user.metadata.lastSignInTime);
      let callback;
      if (isNewUser) {
        callback = onCreateAccount;
      } else if (isRecovery) {
        callback = onSignIn;
      } else {
        callback = onCreateAccount;
      }

```

so its not fully refactor-able

Reply:
Cool! No problem then, let's work on it after merge both PRs.